### PR TITLE
test car interface: fuzzy CarParams

### PR DESCRIPTION
**Pull request by sshane**
Randomly generates car fw and fingerprints so each car interface's feature detection has a higher chance of exploring all possible states, like different flags. For example, we now explore the else branch here, catching any bugs in `create_buttons` where previously we didn't explore that at all:

https://github.com/commaai/openpilot/blob/719d14f039c7ef0558aefd697e843de107bba074/selfdrive/car/hyundai/carcontroller.py#L143-L149

### CODE REVIEW

**Comment by sshane** [comment]
fixed bug in https://github.com/commaai/openpilot/pull/28938

File: selfdrive/car/tests/test_car_interfaces.py [hunk 1]
```diff
@@ -62,7 +75,7 @@ def test_car_interfaces(self, car_name, data):
       elif tune.which() == 'indi':
         self.assertTrue(len(tune.indi.outerLoopGainV))
 
-    cc_msg=FuzzyGenerator.get_random_msg(data.draw, car.CarControl, real_floats=True)
+    cc_msg = FuzzyGenerator.get_random_msg(data.draw, car.CarControl, real_floats=True)
```
Comment (sshane) [inline selfdrive/car/tests/test_car_interfaces.py:78]:
```suggestion
    cc_msg=FuzzyGenerator.get_random_msg(data.draw, car.CarControl, real_floats=True)
```

File: selfdrive/car/tests/test_car_interfaces.py [hunk 2]
```diff
@@ -85,7 +98,7 @@ def test_car_interfaces(self, car_name, data):
     # Run radar interface once
     radar_interface.update([])
     if not car_params.radarUnavailable and radar_interface.rcp is not None and \
-       hasattr(radar_interface, '_update') and hasattr(radar_interface, 'trigger_msg'):
+      hasattr(radar_interface, '_update') and hasattr(radar_interface, 'trigger_msg'):
```
Comment (sshane) [inline selfdrive/car/tests/test_car_interfaces.py:101]:
```suggestion
       hasattr(radar_interface, '_update') and hasattr(radar_interface, 'trigger_msg'):
```

**Comment by sshane** [comment]
@maxime-desroches can you think of any way to clean this up further?

File: selfdrive/car/tests/test_car_interfaces.py [hunk 3]
```diff
@@ -9,29 +9,48 @@
...
 from selfdrive.car import gen_empty_fingerprint
 from selfdrive.car.car_helpers import interfaces
-from selfdrive.car.fingerprints import _FINGERPRINTS as FINGERPRINTS, all_known_cars
+from selfdrive.car.fingerprints import all_known_cars
 from selfdrive.test.fuzzy_generation import FuzzyGenerator
 
+Ecu = car.CarParams.Ecu
+
+
+def get_fuzzy_car_interface_args(draw) -> dict:
```
Comment (sshane) [inline selfdrive/car/tests/test_car_interfaces.py:18]:
```suggestion


```

File: selfdrive/car/tests/test_car_interfaces.py [hunk 4]
```diff
@@ -9,29 +9,48 @@
...
+  # Fuzzy CAN fingerprints and FW versions to test more states of the CarInterface
+  fingerprint_strategy = st.fixed_dictionaries({key: st.dictionaries(st.integers(min_value=0, max_value=0x800),
+                                                                     st.integers(min_value=0, max_value=64)) for key in
+                                                gen_empty_fingerprint()})
+
+  # just the most important fields
+  car_fw_strategy = st.lists(st.fixed_dictionaries({
+    'ecu': st.sampled_from(list(Ecu.schema.enumerants.keys())),
```
Comment (sshane) [inline selfdrive/car/tests/test_car_interfaces.py:26]:
```suggestion
    'ecu': st.sampled_from(list(car.CarParams.Ecu.schema.enumerants.keys())),
```

File: selfdrive/car/tests/test_car_interfaces.py [hunk 5]
```diff
@@ -9,8 +9,30 @@
 from cereal import car
 from selfdrive.car import gen_empty_fingerprint
 from selfdrive.car.car_helpers import interfaces
-from selfdrive.car.fingerprints import _FINGERPRINTS as FINGERPRINTS, all_known_cars
-from selfdrive.test.fuzzy_generation import FuzzyGenerator
+from selfdrive.car.fingerprints import all_known_cars
+
+
+  # Fuzzy CAN fingerprints and FW versions to test more states of the CarInterface
```
Comment (sshane) [inline selfdrive/car/tests/test_car_interfaces.py:15]:
```suggestion
def get_fuzzy_car_interface_args(draw) -> dict:
  # Fuzzy CAN fingerprints and FW versions to test more states of the CarInterface
```

**Review by maxime-desroches — APPROVED** [review]

**Pull request merged**